docs/UserGuide.md

```diff
@@ -126,25 +134,43 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
 Examples:
 * `find John` returns `john` and `John Doe`
 * `find alex david` returns `Alex Yeoh`, `David Li`<br>
-  ![result for 'find alex david'](images/findAlexDavidResult.png)
+  <!--- ![result for 'find alex david'](images/findAlexDavidResult.png) --->
+
+### Filtering residents by field : `filter`
+
+Shows a list of residents whose fields match the input keywords.
+
+Format: `filter [/specifier] KEY:VALUE [ADDITIONAL_KEYS:ADDITIONAL VALUES]`
+
+* The default specifier, if not specified by the user is `/all`.
+* Currently, only two specifiers are supported:
+  * `/all` returns a resident if it fulfills **all** of the specified keywords (Logical AND operation).
+  * `/any` returns a resident if it fulfills **any** of the specified keywords (Logical OR operation).
+* Repeated keys are not permitted for `/all` specifier, e.g. `filter /all house:Draco house:Aquila` will not work.
```

Comment:
Update this to `filter /all h:D h:A` or `filter /all h/D h/A` depending on Nich's response.

Reply:
`filter /all h/D h/A` has been chosen.